packages/web/src/components/receive-tokens-modal/ReceiveTokensModal.tsx

```diff
@@ -0,0 +1,151 @@
+import { useCallback, useContext } from 'react'
+
+import { useWalletAddresses } from '@audius/common/api'
+import { useReceiveTokensModal } from '@audius/common/store'
+import {
+  Button,
+  Flex,
+  IconError,
+  Text,
+  LoadingSpinner,
+  Divider,
+  Hint
+} from '@audius/harmony'
+import QRCode from 'react-qr-code'
+import { useAsync } from 'react-use'
+
+import { AddressTile } from 'components/address-tile'
+import { CryptoBalanceSection } from 'components/buy-sell-modal/CryptoBalanceSection'
+import { ExternalTextLink } from 'components/link'
+import ResponsiveModal from 'components/modal/ResponsiveModal'
+import { ToastContext } from 'components/toast/ToastContext'
+import { useIsMobile } from 'hooks/useIsMobile'
+import { getUserBank } from 'services/solana/solana'
+import { copyToClipboard } from 'utils/clipboardUtil'
+
+const DIMENSIONS = 160
+
+const messages = {
+  title: 'RECEIVE',
+  explainer: 'Send tokens to your built in Audius wallet.',
+  disclaimer: 'Use caution to avoid errors and lost funds.',
+  learnMore: 'Learn More',
+  copy: 'Copy Wallet Address',
+  close: 'Close',
+  copied: 'Copied to Clipboard!'
+}
+
+export const ReceiveTokensModal = () => {
+  const { isOpen, onClose, data } = useReceiveTokensModal()
+  const { data: walletAddresses } = useWalletAddresses()
+  const { currentUser: wallet } = walletAddresses ?? {}
+  const { toast } = useContext(ToastContext)
+  const isMobile = useIsMobile()
+  const { tokenInfo, balance } = data
+
+  const { value: userBankAddress } = useAsync(async () => {
```

Comment:
in our current implementation of the Receive modal we create a user bank if needed, wonder if we want to do that here? could be costly due to the amount of coins there will be

Reply:
yeah i need to add that back in + i wanted to check with you on an implementation that feels a bit cleaner